### PR TITLE
Feature/update details screens

### LINKED ISSUE #415: Update ShowEventScreen

### LINKED ISSUE #416: Update SerieDetailsScreen

### PR DESCRIPTION
**Pull request by Saifurrhn**
## Description

This is a small PR where the group name is shown in the events and series details screens when it belongs to a group.

## Changes

**ShowEventViewModel and SerieDetailsViewModel**:

- `getGroupName(groupId: String)`: return the group name
- `groupId` and `groupName` added to the state

**ShowEventScreen and SerieDetailsScreen**:

- Show the group name under the owner name

## Screenshots

image

image

## Issues

Closes #415 
Closes #416

### CODE REVIEW

File: app/src/main/java/com/android/joinme/ui/overview/ShowEventScreen.kt [hunk 1]
```diff
@@ -402,6 +405,20 @@ fun ShowEventScreen(
                           .padding(vertical = Dimens.Padding.small),
                   textAlign = TextAlign.Center)
 
+              // Group display (if event belongs to a group)
+              if (eventUIState.groupName != null) {
+                Text(
+                    text = "${stringResource(R.string.group_name)} ${eventUIState.groupName}",
+                    style = MaterialTheme.typography.bodyMedium,
+                    fontWeight = FontWeight.Medium,
```
Comment (MathieuPFR) [inline app/src/main/java/com/android/joinme/ui/overview/ShowEventScreen.kt]: I see a difference here with the fontWeight of SerieDetailsScreen, is this on purpose ? if not maybe stay consistent with both screen for a better user experience

File: app/src/main/java/com/android/joinme/ui/overview/ShowEventViewModel.kt [hunk 2]
```diff
@@ -139,6 +149,24 @@ class ShowEventViewModel(
...
+  private suspend fun getGroupName(groupId: String): String? {
+    if (groupId.isEmpty()) {
+      return null
+    }
+    return try {
+      val group = groupRepository.getGroup(groupId)
+      group.name
+    } catch (_: Exception) {
+      null
+    }
```
Comment (MathieuPFR) [inline app/src/main/java/com/android/joinme/ui/overview/ShowEventViewModel.kt]: same thing here : getGroupName checks empty string and returns null; some places call getGroupName only if groupId != null (which allows empty string through). This is fine but slightly inconsistent.

**Review by MathieuPFR — CHANGES_REQUESTED** [review]
I didn't find any blocking bugs this is a good PR, good job
The feature (displaying group name in Serie/Event screens + fetching it in viewmodels) is implemented sensibly.
I have a set of recommendations to improve localization, reduce test duplication and tighten a few edge-cases but except that I have nothing more to say. So I commented directly in line and maybe I have a few suggestions for the tests below: 

in SereDetailsViewModelTest, you could try to merge loadSerieDetails_withGroupId_groupNotFound_setsGroupNameToNull and loadSerieDetails_withGroupId_repositoryThrows_setsGroupNameToNull since the implementation treats all exceptions identically.

in ShowEventViewModelTest this is essentially the same thing you could merge loadEvent_withGroupId_groupNotFound_setsGroupNameToNull and loadEvent_withGroupId_repositoryThrows_setsGroupNameToNull since the implementation treats all exceptions identically.
and then I think it's also possible to merge the last two test of this file since they check the same behavior. 
This would reduce the number of tests from 13 to 10. Also you have a very nice line coverage, well done !
